pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -159,7 +173,30 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 		orchestrationStatusKey, provisioningLabelKey,
 	)
 
-	originalNodes, remainingNodes, completed, statuses, err := c.checkNodesCompleted(selector, config, configOSDs)
+	// For OSDs that are definitely update operations, we want to be able to defer the update until
+	// after we have created new OSDs in order to speed up cluster scale-out. Allow us to defer
+	// updating by keeping a list of deferred configmaps and processing those after creates have
+	// been done. If OSD provisioning doesn't report a "preexisting" status, then no configmaps will
+	// be deferred, and OSD create/update ops will be intermixed in whatever order is processed.
+	deferredConfigMaps := []*v1.ConfigMap{}
+	defer func() {
+		for _, configMap := range deferredConfigMaps {
+			node, ok := configMap.Labels[nodeLabelKey]
+			if !ok {
+				logger.Infof("missing node label on deferred configmap %s", configMap.Name)
+				continue
+			}
+
+			c.handleDeferredStatusConfigMapStatus(node, config, configMap, configOSDs)
```

Comment:
Will all OSD updates be deferred until after all nodes complete the provisioning? What if a node doesn't respond? Seems like we need to proceed with the OSD updates immediately to avoid getting stuck in corner cases. After each OSD is updated, the operator could check again if there are any new OSDs to provision before continuing with the updates. 

What if all OSD creates or updates were processed from a queue? New OSDs would be added to the head of the queue, while existing OSDs would be added to the end of the queue. And when we have multiple failure domains, it would turn into one queue per failure domain where the operator would process one failure domain at a time.

Reply:
That's a good thought. I should consider that corner case so we don't get stuck.

Reply:
If there are any nodes or new PVs hung during provisioning, it would block all upgrades until the 20 minute timeout has elapsed? That's a pretty severe penalty even if it doesn't commonly happen. Do we need the upodate-in-parallel changes with this PR to work around this?

Reply:
I can work on a system that will work on updates while waiting on CMs to finish, but it'll be more code work. And Testing work I would think also. I'll start on it.